pkg/kube/pod_runner.go

```diff
@@ -26,9 +26,14 @@ import (
 	"github.com/kanisterio/kanister/pkg/log"
 )
 
+type PodOutputMap map[string]interface{}
+
+type PodRunnerFunc func(context.Context, PodController) (PodOutputMap, error)
+
 // PodRunner allows us to start / stop pod, write file to pod and execute command within it
 type PodRunner interface {
 	Run(ctx context.Context, fn func(context.Context, *v1.Pod) (map[string]interface{}, error)) (map[string]interface{}, error)
+	RunEx(ctx context.Context, fn PodRunnerFunc) (PodOutputMap, error)
```

Comment:
Nit. I think we should either make named function types for both - Run and RunEx, or for none. The same is for the returned map type.

Reply:
I think we should add a comment where on why we have to methods that are doing to the same things. Basically what you have explained in the comment below.
Also, I think `RunEx` is not very intuitive, do we have other alternate names :) ?

Reply:
I've added the comment, but don't have better name. If you have something in mind, please suggest.

Reply:
@denisvmedia I made returned type similar for both functions, but was unable to invent name for functor type, so I left it as is.

Reply:
If no one has better names I suggest these:

- `PodAwareRunFunc`
- `PodControllerAwareRunFunc`

Reply:
I am assuming that we are going to get rid of `Run` method eventually. And then we can rename `RunEx` to `Run`.
I am ok with `RunEx` as well.